src/privileged/multipreffer/api.js

```diff
@@ -30,6 +32,9 @@ this.multipreffer = class extends ExtensionAPI {
 
 this.FirefoxHooks = {
   init() {
+    AddonStudies.addUnenrollListener(gExtension.id, async (reason) => {
+      await this.cleanup();
```

Comment:
Prior to this, `cleanup` was only called from the `AddonListener` - which only runs once (it removes itself after calling `cleanup`).

Since we are now calling `cleanup` from two places, we should make the `cleanup` function itself idempotent. One way to do this would be:
1. Introduce a new boolean `_cleanedUp` and return early in `cleanup` if this is true.
2. Set `_cleanedUp` to true at the end of `cleanup`